glib/src/subclass/object.rs

```diff
@@ -224,54 +224,39 @@ mod test {
     use super::super::super::value::{ToValue, Value};
     use super::*;
     use crate::{StaticType, Type};
+    use crate as glib;
 
     use std::cell::RefCell;
 
     mod imp {
         use super::*;
 
         // A dummy `Object` to test setting an `Object` property and returning an `Object` in signals
+        #[derive(Default)]
         pub struct ChildObject;
+
+        #[glib::object_subclass]
         impl ObjectSubclass for ChildObject {
             const NAME: &'static str = "ChildObject";
             type Type = super::ChildObject;
             type ParentType = Object;
-            type Interfaces = ();
-            type Instance = subclass::simple::InstanceStruct<Self>;
-            type Class = subclass::simple::ClassStruct<Self>;
```

Comment:
Does it make these optional but if they're provided nonetheless in the `impl` block it will use those instead?

Reply:
Yes.